src/gui/uithememanager.cpp

```diff
@@ -140,3 +159,80 @@ QString UIThemeManager::getIconPath(const QString &iconId) const
 #endif
     return IconProvider::instance()->getIconPath(iconId);
 }
+
+void UIThemeManager::loadColorsFromJSONConfig()
+{
+    QFile configFile(CONFIG_FILE_NAME);
+    if (!configFile.open(QIODevice::ReadOnly)) {
```

Comment:
Shouldn't `config.json` be optional? As well as `stylesheet.qss`?
IMO, we should not require any part of the Theme to be mandatory. Some theme can provide icons only, some other - stylesheet etc.
The only valid requirement, IMO, is that if the file exists, it must have the correct format.
So if some file doesn't exist we should just skip its handling.

@jagannatharjun, @Chocobo1?

Reply:
>So if some file doesn't exist we should just skip its handling.

Yes, that make sense.

Reply:
linked error messages will help in debugging if someone doesn't intend to omit confg.json or stylesheet.qss

maybe make config.json mandatory and/or based on value on "stylesheet" inside config.json we show the error messages, if its false we dont' logthe error message for the cases when stylesheet doesn't exist else we don't log the error messages.

Reply:
>  based on value on "stylesheet" inside config.json

maybe it can be generalized and based on this flag we actually load the stylesheet